### PR TITLE
libcloud backend

### PR DESCRIPTION
**Pull request by aidangomez**

### CODE REVIEW

File: cloud/envs/aws.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+  def __init__(self, config, **kwargs):
+    super().__init__(**kwargs)
+    self.access_key = config["access_key"]
+    self.secret_key = config["secret_key"]
+
+  @property
+  def driver(self):
+    if getattr(self, '_driver', None) is None:
+      self._driver = get_driver(Provider.EC2)(self.access_key, self.secret_key)
```
Comment (1vn) [inline cloud/envs/aws.py]: Provider.AZURE

with this init scheme
```
driver = cls(subscription_id='subscription-id',
             key_file='/path/to/azure_cert.pem')
```

Comment (1vn) [reply cloud/envs/aws.py]: lol, got confused by the class name. this should work for AWS

File: cloud/envs/aws.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+from cloud.envs import env
+from cloud.envs import registry
+from cloud.envs import utils
+
+
+@registry.register("aws")
+class AzureInstance(env.Instance):
```
Comment (1vn) [inline cloud/envs/aws.py]: aws

File: cloud/envs/aws.py (hunk 1 above)
Comment (aidangomez) [reply cloud/envs/aws.py]: lol oooopps

File: cloud/cloud.py [hunk 3]
```diff
@@ -1,18 +1,29 @@
...
-  if isinstance(arg, str):
+def connect(arg=None):
+  if arg is None:
+    with open(path, "r") as cf:
+      config = toml.loads(cf)
+      provider = config.pop("provider").lower()
+      cloud.instance = reg.retrieve(provider, config=config)
+  elif isinstance(arg, str):
     cloud.instance = reg.retrieve(arg)
   elif isinstance(arg, Instance):
```
Comment (1vn) [inline cloud/cloud.py]: when would this be used?

File: cloud/cloud.py [hunk 4]
```diff
@@ -1,18 +1,29 @@
...
+
 import cloud
 from cloud import registry as reg
 from cloud import Instance
 
 
-def connect(arg):
-
-  if isinstance(arg, str):
+def connect(arg=None):
```
Comment (1vn) [inline cloud/cloud.py]: imo, this should ONLY take path to config

File: cloud/cloud.py (hunk 3 above)
Comment (aidangomez) [reply cloud/cloud.py]: If they have their own custom Instance object they want to use

File: cloud/cloud.py (hunk 4 above)
Comment (aidangomez) [reply cloud/cloud.py]: imho, it should NOT

File: cloud/cloud.py [hunk 5]
```diff
@@ -1,18 +1,24 @@
...
+import toml
+
 import cloud
 from cloud import registry as reg
 from cloud import Instance
 
+instance = None
+
+
+def connect(path):
```
Comment (aidangomez) [inline cloud/cloud.py]: I don't think the use should have to pass a path each time. We should have some hierarchy of where we look for a `cloud.config` file:

1. In $CLOUD_CFG
2. In "/"
3. Passed to connect()

File: configs/cloud.toml-gcp [hunk 6]
```diff
@@ -0,0 +1,3 @@
+provider = "gcp"
+
+# GCP doesn't require any additional keys for authentication
```
Comment (1vn) [inline configs/cloud.toml-gcp]: 🔥

**Pull request merged**